apps/meteor/ee/app/livechat-enterprise/server/hooks/afterRemoveDepartment.ts

```diff
@@ -15,8 +15,14 @@ const afterRemoveDepartment = async (options: {
 
 	const { department } = options;
 
-	cbLogger.debug(`Removing department from forward list: ${department._id}`);
-	await LivechatDepartment.removeDepartmentFromForwardListById(department._id);
+	cbLogger.debug({
+		msg: 'Post removal actions on EE code for department',
+		department,
+	});
+	await Promise.all([
+		LivechatDepartment.removeDepartmentFromForwardListById(department._id),
+		...(department.parentId ? [LivechatUnit.removeDepartmentFromUnit(department.parentId)] : []),
```

Comment:
I'm wondering if we should really use the `LivechatUnit` model here since it adds some extra queries to assure the user who requested this action can really view/maange the unit. In this case, having this validation could cause inconsistencies, couldn't it? I mean in case the user can manage the department, but can't manage its unit

Maybe we should use the `LivechatDepartment` model instead :thinking: wdyt?

Reply:
- Should be on `Units` model, cause you're updating a unit.
- The restrictions on the Units model should be manual. I don't remember applying automatic restrictions to any query (we removed that behavior some time ago) so the `updateOne` should go clean. Unless something has changed that i'm not aware of 👀

Reply:
Just noticed the `addQueryRestrictions` function we have defined in this file is not applied to updateOne, so we should be fine :+1: